src/ProjectFileManager.cpp

```diff
@@ -196,15 +197,8 @@ auto ProjectFileManager::ReadProjectFile(
                   right->SetPan(1.0f);
                   RealtimeEffectList::Get(*left).Clear();
                   RealtimeEffectList::Get(*right).Clear();
-                  if(left->GetRate() != right->GetRate())
-                     //i18n-hint: explains why opened project was auto-modified 
-                     linkTypeChangeReason = XO("This project contained stereo tracks with different sample rates per channel.");
-                  else if(left->GetSampleFormat() != right->GetSampleFormat())
-                     //i18n-hint: explains why opened project was auto-modified  
-                     linkTypeChangeReason = XO("This project contained stereo tracks with different sample formats in channels.");
-                  else
-                     //i18n-hint: explains why opened project was auto-modified 
-                     linkTypeChangeReason = XO("This project contained stereo tracks with non-aligned content.");
+                  if(!linkTypeChangeReason)
+                     linkTypeChangeReason = WaveTrackUtils::GetChannelUnlinkReason(*left, *right);
```

Comment:
However, I see that in case of opening a 3.x file, not importing .aup, the warning about possible loss of stacked effects is always given, in addition to whatever other message is reported by the function.  That is unchanged behavior compared with master.